recommendation/search/recommendation.py

```diff
@@ -16,8 +16,12 @@ class SearchRecommendation(object):
       sequentially determine if they match the result, then enhance the result
       if so.
     """
-    def __init__(self, query):
+    def __init__(self, query, debug=False):
+        self.debug = debug
         self.query = query
+        self.recommendation = None
+        self.suggestion_engine = self.get_suggestion_engine()
+        self.query_engine = self.get_query_engine()
```

Comment:
If you are by-passing the getter methods anyways, maybe you could just make the two engines class level attributes and avoid the getter altogether.